### PR TITLE
Balance tests across CI_NODE_TOTAL using go test -list

### PR DESCRIPTION
**Pull request by brandonc**

### CODE REVIEW

File: helper_test.go [hunk 1]
```diff
@@ -1924,27 +1925,60 @@ func randomSemver(t *testing.T) string {
...
+	result := make(map[string]int)
+	index := 0
+	for _, line := range strings.Split(string(output), "\n") {
+		if strings.HasPrefix(line, "Test") {
+			result[line] = index
+			index += 1
+		}
 	}
-	return "3"
+	return result
```
Comment (sebasslash) [inline helper_test.go:1981]: I just ran `go test -v ./... -list=. -tags=integration` and it is a tad bit slow. Alternatively, we can add the assignment for `testNames` to `init()` so it runs before any tests are run: 
```go
func init() {
  testNames = listTestsCI()
}
```
and remove `t.Helper()`. What do you think?

Comment (brandonc) [reply helper_test.go:1981]: To be clear I should only be running it once per node because it's being assigned to a package variable

Comment (brandonc) [reply helper_test.go:1981]: I think init is a better approach. I'll modify it.

**Review by sebasslash — APPROVED** [review]
FWIW, I **really** like this change since it saves us from having to rebalance tests manually. 

Good with me 🚀 . There's an odd failure going on with the codegen workflow 😅

**Pull request merged**